### PR TITLE
Update to Jest 12. Codemod to new Jest APIs.

### PR DESCRIPTION
**Pull request by cpojer**
There is one test failing right now. Pretty sure it is because we updated to jsdom 8. Will look into this tomorrow unless anyone has some ideas – seems like the multiple defaultValues test is failing in ReactDOMSelect – is this a bug in React or jsdom?

### CODE REVIEW

**Comment by jimfb** [comment]
The unit test expectation matches what the browser does, so it looks like a jsdom bug.

**Comment by zpao** [comment]
This is just the jest update, not the Jasmine 2 update correct?

**Comment by cpojer** [comment]
yes.

Waiting for the jsdom fix here: https://github.com/tmpvar/jsdom/pull/1462/commits/236caaef9fd7a5fe3c7204b8569396da6d37e918

**Comment by facebook-github-bot-4** [comment]
@cpojer updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@cpojer updated the pull request.

File: grunt/tasks/jest.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ function writeTempConfig(callback) {
 function run(done, configPath) {
   grunt.log.writeln('running jest (this may take a while)');
 
-  var args = ['--harmony', path.join('node_modules', 'jest-cli', 'bin', 'jest')];
+  var args = ['--harmony', path.join('node_modules', 'jest-cli', 'bin', 'jest'), '--runInBand'];
```
Comment (cpojer) [inline grunt/tasks/jest.js]: cc @zpao @gaearon I'm gonna ship this with `--runInBand` – it should avoid the failures on travis you guys have been seeing. Let me know if you have any trouble with this.

**Comment by facebook-github-bot-0** [comment]
@cpojer updated the pull request.

**Pull request merged**